opendere/ability.py

```
@@ -24,36 +29,51 @@ def __init__(self, num_uses=0, phases=[], command_public=False):
         self.phases = phases
         self.command_public = command_public
 
-    def __call__(self, apply_immediately, game, user, target_user=None):
-        action_obj = self.action(game, user, target_user)
-        if apply_immediately:
-            action_obj()
+    def __call__(self, game, user, target_user=None):
+        previous_action = next((act for act in game.phase_actions if isinstance(act, self.action) and user == act.user), None)
+        if previous_action and previous_action.target_user == target_user:
+            if self.command_public:
+                return [(game.channel, f"{user}: you've already told me you were going to do that, baka ;_;")]
+            return [(user.uid, f"{user}: you've already told me you were going to do that, baka ;_;")]
+
+        if previous_action:
+            game.phase_actions.remove(previous_action)
+            action_obj = self.action(game, user, target_user, ability=self, previous_action=previous_action)
```

Can the action be invalid? For example an invalid target? If so, `self.action()` should raise an Exception, and that exception should be handled. If there is an exception, it shouldn't remove the previous action.